File: services/gateway/internal/domain/author_profile.go
package domain

type AuthorProfile struct {
	Orcid       string `json:"orcid"`
	Confirmed   bool   `json:"confirmed"`
	ConfirmedAt string `json:"confirmed_at"`
	PaperCount  int64  `json:"paper_count"`
}

type AuthorProfileUpdateInput struct {
	Orcid             string `json:"orcid"`
	ConfirmAuthorship bool   `json:"confirm_authorship"`
}

File: services/semantic/src/domain/models/author_profile.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(slots=True)
class AuthorProfileModel:
    user_id: int
    orcid: str | None = None
    confirmed_at: datetime | None = None
    paper_count: int = 0
    created_at: datetime | None = None
    updated_at: datetime | None = None

    @property
    def confirmed(self) -> bool:
        return bool(self.orcid and self.confirmed_at)
